### ISSUE TITLE
parse result should have more specific type

### ISSUE THREAD

**Issue by NullVoxPopuli** [embroider-build/content-tag#30]
right now the type is `any`. A better type might be something like:
```ts
interface ParseResult {
  type: "expression" | "class-member",
  tagName: "template",
  contents: string,
  range: {
    start: number,
    end: number,
  },
  contentRange: {
    start: number,
    end: number,
  },
  startRange: {
    end: number,
    start: number,
  },
  endRange: {
    start: number,
    end: number,
  },
}

// 
let result: ParseResult[] = p.parse(...);
```

**Comment by ef4** [comment]
There's a [merged but unreleased PR](https://github.com/rustwasm/wasm-bindgen/pull/3554) in wasm-bindgen that would let us return `Result<Vec<Occurence>, JsValue>` directly across the wasm boundary. That would probably cause wasm-bindgen to also generate this type better.

**Comment by NullVoxPopuli** [comment]
this is resolved as of 1.2.2
